nsv13/code/modules/overmap/weapons.dm

```diff
@@ -168,32 +168,60 @@
 			fire_railgun(target)
 		if(FIRE_MODE_TORPEDO) //In case of bugs.
 			fire_torpedo(target)
+		if(FIRE_MODE_LASER)
+			fire_laser(target)
 
 /obj/structure/overmap/verb/cycle_firemode()
 	set name = "Switch firemode"
 	set category = "Ship"
 	set src = usr.loc
 	if(usr != gunner)
 		return
-	var/max_firemode = FIRE_MODE_RAILGUN
+	var/max_firemode = FIRE_MODE_LASER
 	if(mass < MASS_MEDIUM) //Small craft dont get a railgun
 		max_firemode = FIRE_MODE_TORPEDO
-	fire_mode ++
-	if(fire_mode > max_firemode)
-		fire_mode = FIRE_MODE_PDC
+
+	var/tries = 0
+	while (tries < 4) // The total number of weapon types
```

Comment:
I recommend you add a:
stoplag() 
in the scope of this while loop, or else you could probably spam CTRL with a script to cause a lot of while loops and...you see where this is going

Reply:
Alternatively, a cooldown would also work.

Reply:
I put it at the beginning hopefully that's right